app/frontend/wallet/account-manager.ts

```diff
@@ -0,0 +1,65 @@
+import NamedError from '../helpers/NamedError'
+import {Account} from './account'
+import {CryptoProviderFeatures, MAX_ACCOUNT_COUNT} from './constants'
+
+const AccountManager = ({config, cryptoProvider, blockchainExplorer}) => {
+  const accounts: Array<ReturnType<typeof Account>> = []
+
+  function getAccount(accountIndex: number) {
+    return accounts[accountIndex]
+  }
+
+  function discoverNextAccount() {
+    return Account({
+      config,
+      cryptoProvider,
+      blockchainExplorer,
+      accountIndex: accounts.length,
+    })
+  }
+
+  async function addNextAccount(account) {
+    await account.ensureXpubIsExported() // To ensure user exported pubkey
+    const isLastAccountUsed =
+      accounts.length > 0 ? await accounts[accounts.length - 1].isAccountUsed() : true
+    if (
+      account.accountIndex !== accounts.length ||
+      !isLastAccountUsed ||
+      account.accountIndex > MAX_ACCOUNT_COUNT
```

Comment:
given this condition, MAX_ACCOUNT_COUNT is kind of a misleading name, MAX_ACCOUNT_INDEX would be more accurate

Reply:
also, if this condition errors, this check should actually be in discoverAccounts, still it's probably not a bad idea to keep it here as a safeguard

Reply:
and finally, I would add maxAccountCount parameter to AccountManager which by default would be  set to MAX_ACCOUNT_COUNT, to be able to set it to something lower from outside to simplify unit testing